src/ide/tgui.ts

```diff
@@ -1146,14 +1167,401 @@ export let tgui = (function() {
+	//                      Return true if the dialog should be kept opened, otherwise return false or nothing at all.
+	//                      
+	//
+	// those properties are carried over to the returned object
+	// and the following fields are contained in the returned object:
+	// - content:           a DOM element, that represents the content of the dialog
+	// - dom:               a DOM element, that represents the whole dialog
+	// - button_doms:       dictionary of DOM elements, that represents the buttons in the button bar
+	// - and others, mainly the titlebar components
+	//
+	// TODO: Support undecorated elements via a boolean property `decorated`, in this case
+	//       control.dom might be the same as control.content.
+	module.createModal = function(description)
+	{
+		// control -- description object returned from this function and stored in the tgui.modal array
+		// dialog  -- the DOM object of the dialog contents
+		let control = Object.assign({}, description);
+		// handle default fields
+		if(!control.hasOwnProperty("onButton")) control.onButton = function(button) {};
+
+		let handleChoice = function(event, button)
+		{	
+			// if the onButton returns true, the dialog is kept opened
+			let ret = control.onButton(button);
+			if(event)
+			{
+				event.preventDefault();
+				event.stopPropagation();
+			}
+			if(!ret) tgui.stopModal(control); // close current dialog
+			return false;
+		}
+		
+		// create dialog	
+		let dialog = tgui.createElement({
+			"type": "div",
+			"className": "tgui tgui-modal",
+			"style": {"background": "#eee", "overflow": "hidden", "display": "block", "zIndex": 100},
+		});
+		control.dom = dialog;
+		
+		dialog.addEventListener("keydown", function(event)
+		{
+			if(event.keyCode === 9) // TAB characterSet
+			{			
+				//let focus_query = 'button, [href], input, select, textarea, [tabIndex]:not([tabIndex="-1"])';
+				let focus_query = 'button, [href], input, select, textarea';
+				// collect all focusable elements inside the dialog, except titlebar buttons
+				var focusable = Array(...control.content.querySelectorAll(focus_query));
+				if(control.hasOwnProperty("div_buttons"))
+				{
+					focusable.push(...control.div_buttons.querySelectorAll(focus_query));
+				}			
+				
+				if(focusable.length === 0)
+				{
+					// Nothing inside the dialog focusable, so keep focus away from other elements
+					(document.activeElement as any)?.blur();
+				}
+				else
+				{
+					let focusIndex = focusable.indexOf(document.activeElement);
+					let n = focusable.length;
+					if(focusIndex == -1 || (!event.shiftKey && focusIndex == n-1))
+					{
+						focusable[0].focus();
+						event.preventDefault();
+					}
+					else if(event.shiftKey && focusIndex === 0)
+					{
+						focusable[n-1].focus();
+						event.preventDefault();
+					}
+				}
+			}
+		});
+
+		control.handleClose = (event) => handleChoice(event, "defaultClose");
```

Comment:
```suggestion
		control.handleClose = (event) => handleChoice(event, null);
```

Here, I would simply pass `null`, because _no button_ was chosen and `null` won't collide with button names.